### DISCUSSION TITLE
Websocket compression support

### DISCUSSION THREAD

**Discussion by mohzah** [vernemq/vernemq#2102] [Ideas]
Since Cowboy 2.x supports compression
https://ninenines.eu/docs/en/cowboy/2.9/guide/ws_protocol/
Is it possible to support websocket compression.

Related Issue https://github.com/vernemq/vernemq/issues/607

**Comment by ioolkos** [comment]
@mohzah Thanks for transfering this to a discussion!
From your link, it really looks like we could just enable that and the rest would be automatic. I wasn't aware of that. Does this mean we wouldn't even need this as a configurable setting? or would it need to be settable in the `vernemq.conf`?

**Reply by mohzah** [reply]
@ioolkos I am not sure but take a look at this page.
https://ninenines.eu/docs/en/cowboy/2.9/manual/cowboy_websocket/

> **compress (false)**
>    Whether to enable the Websocket frame compression extension. Frames will only be compressed for the clients that support this extension

**Reply by ioolkos** [reply]
We should actually pass this option down to the Websocket layer already! 
https://github.com/vernemq/vernemq/blob/ff29c9400427b69a617b45546feb8d70457c4ff3/apps/vmq_server/src/vmq_websocket.erl#L107

Apologies :) This already seems to be fully automatic and in. Does anyone see this feature *not* working? is there a way to test for you?
